### PR TITLE
Heap space

### PR DESCRIPTION
**Pull request by sbracellari**
Check for previously set allocation of heap space

##### Checklist

-   [x] the [individual contributor license agreement][] is signed
-   [x] commit message follows [commit guidelines][]
-   [x] tests are included
-   [ ] documentation is changed or added
-   [ ] [message properties][] have been updated with new phrases
-   [x] view conforms with [WCAG 2.0 AA][]

##### Description of change

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#commit
[message properties]: https://github.com/Jasig/uPortal/tree/master/uportal-war/src/main/resources/properties/i18n
[WCAG 2.0 AA]: https://www.w3.org/WAI/WCAG20/quickref/?levels=aaa&technologies=smil%2Cpdf%2Cflash%2Csl

### CODE REVIEW

File: etc/tomcat/bin/setenv.bat [hunk 1]
```diff
@@ -22,6 +20,22 @@ if not errorlevel 1 (
     set CATALINA_OPTS=%CATALINA_OPTS% -XX:+UseG1GC
 )
 
+rem Check if heap space allocation is already set
+rem If it is, leave it be
+echo.%CATALINA_OPTS%|findstr /C:"-Xms"
+if not errorlevel 1 (
+    echo Already declared a garbage collector
```
Comment (drewwills) [inline etc/tomcat/bin/setenv.bat]: "garbage collector?"

File: etc/tomcat/bin/setenv.bat [hunk 2]
```diff
@@ -22,6 +20,22 @@ if not errorlevel 1 (
...
+) else (
+    set CATALINA_OPTS=%CATALINA_OPTS% -Xms64m
+)
+
+echo.%CATALINA_OPTS%|findstr /C:"-Xmx"
+if not errorlevel 1 (
+    echo Already declared a garbage collector
```
Comment (drewwills) [inline etc/tomcat/bin/setenv.bat]: "garbage collector?"

**Comment by drewwills** [comment]
Welcome @sbracellari!

**Comment by sbracellari** [comment]
@ChristianMurphy thank you 😃

**Comment by drewwills** [comment]
These defaults are terrible.  Anyone who overlooks settings real values for these JVM args in a server deployment will run into problems very quickly.

My best guess right now is that both `-Xms` & `-Xmx` should be 2 GB, as a default.

**Comment by ChristianMurphy** [comment]
@drewwills PRs setting better defaults are welcome :slightly_smiling_face:

**Pull request merged**